freckle-exception/library/Freckle/App/Exception/MonadThrow.hs

```diff
@@ -97,25 +102,38 @@ try action = withFrozenCallStack $ Annotated.try action
 
 tryJust
   :: forall e b m a
-   . (Exception e, MonadCatch m, HasCallStack)
+   . (Exception e, MonadCatch m)
   => (e -> Maybe b)
   -> m a
   -- ^ Action to run
   -> m (Either b a)
 tryJust test action =
```

Comment:
Couldn't this have been written in terms of `catchJust`, so we only have to fix this bug once?

```hs
tryJust test action = catchJust test (Right <$> action) $ pure . Left
```

?